pkg/state/state_ofed.go

```diff
@@ -590,3 +593,69 @@ func (s *stateOFED) setEnvFromClusterWideProxy(cr *mellanoxv1alpha1.NicClusterPo
 		)
 	}
 }
+
+// mergeWithDefaultEnvs returns env variables provided in currentEnvs merged with default
+// env variables for MOFED container, taking into account attributes of the k8s cluster.
+func (s *stateOFED) mergeWithDefaultEnvs(
+	currentEnvs []v1.EnvVar, nodeAttrs map[nodeinfo.AttributeType]string) []v1.EnvVar {
+	if envVarsWithGet(currentEnvs).Get("CREATE_IFNAMES_UDEV") != nil {
```

Comment:
nit: this construction is nice, but IMO creates unneeded complexity

maybe replace it with dumb loop?
```
	for _, v := range currentEnvs {
		if v == "CREATE_IFNAMES_UDEV" {
			// already exists don't overwrite
			return currentEnvs
		}
	}
```

Reply:
was just thinking  its a matter of time until we need to handle additional vars. id prefer to keep as is  this one.

Reply:
ack